ui/ceval/src/protocol.ts

```diff
@@ -178,22 +164,13 @@ export class Protocol {
       this.currentEval = undefined;
       this.expectedPvs = 1;
 
-      this.setOption(
-        'UCI_Variant',
-        this.work.variant === 'antichess'
-          ? 'giveaway' // for old asmjs fallback
-          : lichessRules(this.work.variant),
-      );
+      this.setOption('UCI_Variant', this.variantMap?.(this.work.variant) ?? this.work.variant);
       this.setOption('Threads', this.work.threads);
       this.setOption('Hash', this.work.hashSize || 16);
       this.setOption('MultiPV', Math.max(1, this.work.multiPv));
 
       this.send(['position fen', this.work.initialFen, 'moves', ...this.work.moves].join(' '));
-      this.send(
-        this.work.maxDepth >= 99
-          ? `go depth ${maxStockfishPlies}` // 'go infinite' would not finish even if entire tree search completed
-          : 'go movetime 90000',
-      );
+      this.send(`go depth 99`);
```

Comment:
there are legit endgame positions where reaching `245` (formerly `maxStockfishPlies`) is viable

Reply:
Does stockfish ever discover something new beyond 99? sometimes it throws assertions in movepick.cpp once it starts chewing through the high hundreds after finding mate. If you want it the cap back 245, I could use your help on that.

Reply:
while i try to find an example position, do you have one for the crash?

in any case, i don't think we have to block the pr on this.

Reply:
Yes, i'll send it to you on zulip

Reply:
this reminded me of a detail that was problematic in earlier ports: stack overflows. the fix was to explicitly set the stack size of search threads via https://github.com/lichess-org/stockfish.wasm/blob/051d2b8268a5df1fcd8eb1eab26918729498b239/src/thread_win32_osx.h#L35. not sure if any of the defaults have changed, in the mean-time.

Reply:
My understanding is -sDEFAULT_PTHREAD_STACK_SIZE linker argument obviates the need for manual calls to pthread_attr_setstacksize. When no value is given for that arg, it claims to default to -sSTACK_SIZE which we do specify at 2MB. I don't think it's the stack though.

Reply:
Look like emscripten has improved :+1: